pennylane/gradients/pulse_generator_gradient.py

```diff
@@ -412,8 +411,263 @@ def processing_fn(results):
+        t = [0.1, 0.9]
+
+    For simplicity, consider a pulse program consisting of this single pulse and a
+    measurement of the expectation value of :math:`X_{0}`.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=2)
+
+        @qml.qnode(dev, interface="jax", diff_method=qml.gradients.pulse_generator)
+        def circuit(params):
+            op = qml.evolve(H)(params, t)
+            return qml.expval(qml.PauliX(0))
+
+    We registered the ``QNode`` to be differentiated with the ``pulse_generator`` method.
+    This allows us to simply differentiate it with ``jax.grad``, which internally
+    makes use of the pulse generator parameter-shift method.
+
+    >>> jax.grad(circuit)(params)
+    [Array(1.4189798, dtype=float32, weak_type=True),
+     Array([0.00164918, 0.00284802], dtype=float32),
+     Array(-0.09984542, dtype=float32, weak_type=True)]
+
+    Alternatively, we may apply the transform to the tape of the pulse program, obtaining
+    the tapes with inserted ``PauliRot`` gates together with the post-processing function:
+
+    >>> circuit.construct((params,), {}) # Build the tape of the circuit.
+    >>> tapes, fun = qml.gradients.pulse_generator(circuit.tape, argnums=[0, 1, 2])
+    >>> len(tapes)
+    12
+
+    Why are there :math:`12` tapes?
+    Consider the terms in the time-dependent pulse Hamiltonian: :math:`\{Y_0, Y_1, Z_0X_1\}`
+    Via the Lie bracket, which is just the standard matrix commutator, they
+    generate an algebra, the so-called *dynamical Lie algebra (DLA)* of the pulse.
+    In order to find all Pauli words that occur in the DLA, we need to (recursively)
+    calculate all possible commutators between the three words above and their
+    commutators. For the three words above, we obtain three additional words:
+
+    .. math::
+
+        [Y_0, Z_0X_1] &\propto X_0X_1 \\
+        [Y_1, Z_0X_1] &\propto Z_0Z_1 \\
+        [Y_0, Z_0Z_1] &\propto X_0Z_1
+
+    All other commutators result in expressions proportional to one of the six Pauli words.
+    For each of these six words, we need to compute the standard parameter-shift rule
+    requiring two shifted circuits, which yields :math:`12` tapes.
+
+    We may inspect one of the tapes, which differs from the original tape by the inserted
+    rotation gate ``"RIY"``, i.e. a ``PauliRot(np.pi/2, "IY", wires=[0, 1])`` gate.
+    Note that the order of the tapes follows lexicographical ordering of the inserted
+    Pauli rotations, so that :math:`Y_1` is the first of the six words.
+
+    >>> print(qml.drawer.tape_text(tapes[0]))
+    0: ─╭RIY─╭ParametrizedEvolution─┤  <X>
+    1: ─╰RIY─╰ParametrizedEvolution─┤
+
+    Executing the tapes and applying the post-processing function to the results then
+    yields the gradient:
+
+    >>> fun(qml.execute(tapes, dev))
+    (Array(1.41897933, dtype=float64),
+     Array([0.00164914, 0.00284789], dtype=float64),
+     Array(-0.09984585, dtype=float64))
+
+    .. note::
```

Comment:
I would move the note about JAX/JIT up above the example section, since I think a lot of people don't make it all the way to the end of the example, and it seems important